src/main/java/fi/metatavu/metaform/server/rest/RealmsApiImpl.java

```diff
@@ -570,4 +912,29 @@ private String serializeMetaform(Metaform metaform) {
     return null;
   }
 
+  /**
+   * Validates incoming Metaform
+   * 
+   * @param payload metaform data
+   * @return validation error or null if metaform is valid
+   */
+  private Response validateMetaform(Metaform payload) {
+    List<String> keys = payload.getSections().stream()
+      .map(MetaformSection::getFields)
+      .flatMap(List::stream)
+      .map(MetaformField::getName)
+      .collect(Collectors.toList());
+    
+    List<String> duplicates = keys.stream()
+      .filter(key -> Collections.frequency(keys, key) > 1)
+      .distinct()
+      .collect(Collectors.toList());
+    
+    if (duplicates.size() > 0) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use isEmpty() to check whether the collection is empty or not. ([squid:S1155](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS1155))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.metaform-api%3Ametaform-api&pullRequest=49&issues=AWWBxMGxXKENQn1S5jLW&open=AWWBxMGxXKENQn1S5jLW)

[ISSUE_KEY]: AWWBxMGxXKENQn1S5jLW